### PR TITLE
motion correction

### PR DESCRIPTION
**Pull request by samuelgarcia**
This PR
* fix some bugs in motion_estimation (time horizon)
* refactor correct_motion on trace with many option (border, interpolation type...)
* try some clean on motion vector.

### CODE REVIEW

**Comment by yger** [comment]
@samuelgarcia  I updated the branch with master, just to double check everything is fine. I'll do more tests, but kriging seems to be slightly better than idw. So my 2 cents is that the best is monopolar + kriging indeed, for sorting purposes. I'll relaunch everything next weeks to prepare nice notebooks

**Review by alejoe91 — APPROVED** [review]
beautiful @samuelgarcia

File: spikeinterface/sortingcomponents/motion_correction.py [hunk 1]
```diff
@@ -188,63 +190,61 @@ def _get_closest_ind(array, values):
...
+                 border_mode='remove_channels', spatial_interpolation_method='kriging',
+                 sigma_um=20., p=1, num_closest=3):
+        assert recording.get_num_segments() == 1, 'correct_motion() is only availabl for single-segment recordings'
         
         channel_locations = recording.get_channel_locations()
 
         spatial_interpolation_kwargs = dict(sigma_um=sigma_um, p=p, num_closest=num_closest)
         if border_mode == 'remove_channels':
+            assert channel_locations.ndim >= direction, (f"'direction' {direction} not available. "
+                                                         "Channel locations have {channel_locations.ndim} dimensions.")
```
Comment (samuelgarcia) [inline spikeinterface/sortingcomponents/motion_correction.py:247]: @alejoe91 this can be outside the if no ?

Comment (alejoe91) [reply spikeinterface/sortingcomponents/motion_correction.py:247]: yep, one sec!

**Pull request merged**